src/base/ddl_parser.cc

```diff
@@ -304,29 +308,40 @@ bool IndexMapBuilder::UpdateIndex(const hybridse::vm::Range& range) {
     }
 
     auto frame = range.frame();
-    auto start = frame->GetHistoryRangeStart();
-    auto rows_start = frame->GetHistoryRowsStart();
+    auto start = -1 * frame->GetHistoryRangeStart();
+    auto rows_start = frame->GetHistoryRowsStartPreceding();
 
-    DLOG_ASSERT(start <= 0 && rows_start <= 0);
+    DCHECK(start >= 0 && rows_start >= 0);
 
     std::stringstream ss;
     range.frame()->Print(ss, "");
     DLOG(INFO) << "frame info: " << ss.str() << ", get start points: " << start << ", " << rows_start;
 
     auto ttl_st_ptr = index_map_[latest_record_];
     auto type = frame->frame_type();
-    if (type == hybridse::node::kFrameRows) {
-        // frame_rows is valid
-        DLOG_ASSERT(frame->frame_range() == nullptr && frame->GetHistoryRowsStartPreceding() > 0);
-        ttl_st_ptr->set_lat_ttl(frame->GetHistoryRowsStartPreceding());
-        ttl_st_ptr->set_ttl_type(type::TTLType::kLatestTime);
-    } else {
-        // frame_range is valid
-        DLOG_ASSERT(type != hybridse::node::kFrameRowsMergeRowsRange) << "merge type, how to parse?";
-        DLOG_ASSERT(frame->frame_rows() == nullptr && frame->GetHistoryRangeStart() < 0);
-        // GetHistoryRangeStart is negative, ttl needs uint64
-        ttl_st_ptr->set_abs_ttl(ConvertToMinute(-1 * frame->GetHistoryRangeStart()));
-        ttl_st_ptr->set_ttl_type(type::TTLType::kAbsoluteTime);
+    switch (type) {
+        case hybridse::node::kFrameRows:
+            DCHECK(frame->frame_rows() && frame->frame_range() == nullptr);
+            ttl_st_ptr->set_lat_ttl(rows_start);
+            ttl_st_ptr->set_ttl_type(type::TTLType::kLatestTime);
+            break;
+        case hybridse::node::kFrameRange:
+        case hybridse::node::kFrameRowsRange:
+            DCHECK(frame->frame_rows() == nullptr && frame->frame_range());
+            // GetHistoryRangeStart is negative, ttl needs uint64
+            ttl_st_ptr->set_abs_ttl(ConvertToMinute(start));
+            ttl_st_ptr->set_ttl_type(type::TTLType::kAbsoluteTime);
+            break;
+        case hybridse::node::kFrameRowsMergeRowsRange:
+            DCHECK(frame->frame_rows() && frame->frame_range());
+            // use abs and ttl, only >abs_ttl and > lat_ttl will be expired
+            ttl_st_ptr->set_lat_ttl(rows_start);
+            ttl_st_ptr->set_abs_ttl(ConvertToMinute(start));
```

Comment:
so the `abs_ttl` of `column_key` is minute?

Reply:
After #597, it already does.

Reply:
https://github.com/4paradigm/OpenMLDB/blob/main/src/cmd/sql_cmd.h#L1020 delete convert in HandleDeploy

Reply:
ok